client/src/lib/src/utils/files.ts

```diff
@@ -5,6 +5,12 @@
 
 import path from 'path'
 
+export const bitbakeFileExtensions = ['.bb', '.bbappend', '.bbclass', '.conf', '.inc']
+
 export function extractRecipeName (filePath: string): string {
-  return path.parse(filePath).name.split('_')[0]
+  // Ex: gst1.0-plugins-bad_1.18.4.bb -> gst1.0-plugins-bad
+  const baseName = path.parse(filePath).base
+  const fileName = baseName.replace(new RegExp(bitbakeFileExtensions.join('|'), 'g'), '')
+  const recipeName = fileName.split('_')[0]
+  return recipeName
```

Comment:
According to the node [docs](https://nodejs.org/api/path.html#pathextnamepath) on `path`. The example you put should produce the intended extension with the original function logic.

Reply:
One thing we could add is to check if the file has `.bb` before extracting the recipe name.

Reply:
This function is sometimes called with gstreamer1.0.bb, sometimes with gstreamer1.0
In which case we both want gstreamer1.0 to be the result. That's why I used explicitely the file extensions like .bb

Reply:
Updated the regex to match extensions only at the end of the string